contracts/adapters/KycOnboarding.sol

```diff
@@ -0,0 +1,267 @@
+import "../guards/AdapterGuard.sol";
+import "../utils/Signatures.sol";
+import "../helpers/WETH.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+MIT License
+
+Copyright (c) 2021 Openlaw
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
+contract KycOnboardingContract is AdapterGuard, Signatures {
+    using Address for address payable;
+    using SafeERC20 for IERC20;
+
+    event Onboarded(DaoRegistry dao, address member, uint256 units);
+
+    struct Coupon {
+        address kycedMember;
+    }
+
+    string public constant COUPON_MESSAGE_TYPE = "Message(address kycedMember)";
+    bytes32 public constant COUPON_MESSAGE_TYPEHASH =
+        keccak256(abi.encodePacked(COUPON_MESSAGE_TYPE));
+
+    bytes32 constant SignerAddressConfig =
+        keccak256("kyc-onboarding.signerAddress");
+
+    bytes32 constant ChunkSize = keccak256("kyc-onboarding.chunkSize");
+    bytes32 constant UnitsPerChunk = keccak256("kyc-onboarding.unitsPerChunk");
+    bytes32 constant MaximumChunks = keccak256("kyc-onboarding.maximumChunks");
+    bytes32 constant MaximumUnits =
+        keccak256("kyc-onboarding.maximumTotalUnits");
+    bytes32 constant MaxMembers = keccak256("kyc-onboarding.maxMembers");
+    bytes32 constant FundTargetAddress =
+        keccak256("kyc-onboarding.fundTargetAddress");
+
+    uint256 private _chainId;
+    WETH private _weth;
+    IERC20 private _weth20;
+
+    event CouponRedeemed(
+        address daoAddress,
+        uint256 nonce,
+        address authorizedMember,
+        uint256 amount
+    );
+
+    struct OnboardingDetails {
+        uint88 chunkSize;
+        uint88 numberOfChunks;
+        uint88 unitsPerChunk;
+        uint88 unitsRequested;
+        uint88 maximumTotalUnits;
+        uint160 amount;
+    }
+
+    mapping(DaoRegistry => uint256) public totalUnits;
+
+    constructor(uint256 chainId, address payable weth) {
```

Comment:
@adridadou this is one thing that was recently refactored for the onboarding adapters. We are now passing the `chainId` and additional cofnigs via `configureDao` function. In order to keep this in sync with other adapter do you think we could do the same here?